app/models/user.rb

```diff
@@ -106,6 +106,50 @@ def gpo_verification_pending_profile?
     gpo_verification_pending_profile.present?
   end
 
+  def suspended?
+    suspended_at.present?
+  end
+
+  def reinstated?
+    reinstated_at.present?
+  end
```

Comment:
To throw a wrench in the works here... 😬  I think these need to consider which state is most current.

As written, an account which was suspended and then reinstated would return true for both `suspended?` and `reinstated?`. It should return false for `suspended?` because the `suspended_at` timestamp is older than the `reinstated_at` timestamp.

Less likely to come up, but it's possible a reinstated account could be re-suspended, so the same logic should see that the `suspended_at` is newer than the `reinstated_at` timestamp.

Reply:
> but it's possible a reinstated account could be re-suspended
Ah, okay. Yes, I didn't come across this in the logic. I'll make the necessary changes.